google/auth/pluggable.py

```diff
@@ -155,7 +174,15 @@ def retrieve_subject_token(self, request):
                 try:
                     # If the cached response is expired, _parse_subject_token will raise an error which will be ignored and we will call the executable again.
                     subject_token = self._parse_subject_token(response)
+                    if (
+                        self._interactive and "expiration_time" not in response
```

Comment:
Why non-interactive mode does not treat missing `expiration_time` as expired?

Reply:
For non-interactive mode, we already do a sanity check in the `_parse_subject_token` for `expiration_time` if an output file is specified. We treat missing expiration time as malformed and throw an exception in case of an output filed is specified.

Reply:
Hi @sai-sunder-s , I changed the check of "expiration_time" and unified the logic between non-interactive and interactive. This will case less confusion to users.
1. Make it as optional regardless of interactive or non-interactive
2. Treat missing expiration time as "expired" when retrieving from file

The reason we want it being optional is:
It doesn't make sense to provide an expiration time when user don't want to cache it regardless interactive or non-interactive.

Reply:
Will this make it less confusing? If we just ignore an invalid output file in non-interactive without throwing an error, the users may not realize that their output file is not being used even if they are creating it. It doesn't make sense with the current implementation for a non-interactive mode output file to not have an expiration time.

Reply:
Just need to document a warning "Missing expiration time will cause the file not being used for caching".
This will keep the interactive and non-interactive treat "expiration_time" in the same way.

Reply:
That would also either:
1: create a difference between the non-interactive mode behavior of the python library and the other 3 already released SDK's
2: Make the behavior of the other 3 SDK's not really make sense if we change them to match, then they wouldn't be checking for expiration_time even though they don't support interactive mode.

Reply:
OK. Make sense. I forget we have other libraries already made that check.